Source/WebCore/platform/network/mac/ResourceHandleMac.mm

```diff
@@ -372,18 +375,23 @@ static bool synchronousWillSendRequestEnabled()
     }
 
     bool shouldUseCredentialStorage = storedCredentialsPolicy == StoredCredentialsPolicy::Use;
+
+    auto contentSniffingPolicy = ContentSniffingPolicy::DefaultForPlatform;
+    if (handle->contentSniffingPolicy() == ContentSniffingPolicy::Disable || (context && context->localFileContentSniffingPolicy() == ContentSniffingPolicy::Disable))
+        contentSniffingPolicy = ContentSniffingPolicy::Disable;
```

Comment:
You use this three-line pattern a few times. It might be nice to make a static helper function for this logic, maybe something like `contentSniffingPolicyForHandleAndContext` or something (here and line 251).